src/vs/code/node/cliVerboseLogger.ts

```diff
@@ -0,0 +1,49 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ChildProcess } from 'child_process';
+import { watch, writeFileSync, unlinkSync } from 'fs';
+import { Promises } from 'vs/base/node/pfs';
+
+/**
+ * This class provides logging for verbose commands on Mac
+ * as part of #102975
+ */
+export class CliVerboseLogger {
+	constructor(readonly filename: string) {
+		writeFileSync(filename, '');
+	}
+
+	public track(): (child: ChildProcess) => Promise<void> {
+		return async (child: ChildProcess) => {
+			const watcher = watch(this.filename);
```

Comment:
Consider to use our built in library `watchFile` for this: `src/vs/base/node/watcher.ts`

I don't mind if we enrich the `src/vs/base/node/watcher.ts` with an option to have a listener on content change. I think that would allow us to inline most of this code into the library function and allow others to use it as well. If you want I can prototype that, or you go ahead.

Reply:
I think that listener already handles content changes. I'll push up a new PR using it soon